update(mysql): getting started guide improvements (#215)
aiven/aiven-docs: docs/products/mysql/get-started.md

   organization > project > Aiven for MySQL service.
1. On the <ConsoleLabel name="overview"/> page of your service, click
   **Quick connect**.
1. In the **Connect** window, select a tool or language to connect to your service, follow
   the connection instructions, and click **Done**.

   ```sql
   mysql --user avnadmin --password=ADMIN_PASSWORD --host mysql-sakila-dev-sandbox.f.aivencloud.com --port 12691 defaultdb
   ```

</TabItem>
<TabItem value="2" label="Terraform">

Access your new service with the MySQL client using the `mysql_service_uri` output you
received after running `terraform apply --auto-approve`.

```sql
mysql "$(terraform output -raw mysql_service_uri)"
```

</TabItem>
<TabItem value="3" label="mysql">
[Connect to your new service](/docs/products/mysql/howto/connect-from-cli) with
[mysql](https://dev.mysql.com/doc/refman/8.0/en/mysql.html).
</TabItem>
</Tabs>

:::tip
Check more tools for connecting to Aiven for MySQL in
[Connect to Aiven for MySQL](/docs/products/mysql/howto/list-code-samples).
:::

## Load a test dataset

`Sakila` is a sample dataset that represents a DVD rental store. It provides a standard
schema highlighting MySQL features.

> 1.  Download the `sakila-db.tar.gz` file from the
>     [MySQL example databases](https://dev.mysql.com/doc/sakila/en/sakila-installation.html)
>     page and unzip it.

Reviewer: ```suggestion
1.  Download the `sakila` database archive (`tar` or `zip` format) from the
    [MySQL example databases](https://dev.mysql.com/doc/sakila/en/sakila-installation.html)
   page and extract it to your desired location (e.g. `/tmp/`).
```

> 1. From the folder where you unzipped the file,

Reviewer: ```suggestion
1. From the folder where you unpacked the archive,
```

sidebar_label: Get started
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import ConsoleLabel from "@site/src/components/ConsoleIcons"

Start using Aiven for MySQL® by creating a service, connecting to it, and loading sample data.

## Prerequisites

- Access to the [Aiven Console](https://console.aiven.io)
- [MySQL CLI client](https://dev.mysql.com/doc/refman/8.0/en/mysql.html)
  installed
- [Terraform installed](https://developer.hashicorp.com/terraform/install) if you prefer
  to get started using code

## Create a service

<Tabs groupId="group1">
<TabItem value="1" label="Console" default>
[Create an Aiven for MySQL® service](/docs/platform/howto/create_new_service)  in the
[Aiven Console](https://console.aiven.io).
</TabItem>
<TabItem value="2" label="Terraform">

1. [Create an authentication token](/docs/platform/howto/create_authentication_token).
1. Store the authentication token in an environment variable:

   ```bash
   export TF_VAR_aiven_api_token=YOUR_AIVEN_API_TOKEN
   ```

1. Create the following Terraform files:

>    - ``provider.tf``, where you specify the version in the ``required_providers`` block

Reviewer: ```suggestion
   - ``provider.tf``, for ``aiven`` provider configuration.
```

      ```hcl
      terraform {
        required_providers {
          aiven = {
            source  = "aiven/aiven"
            version = ">=4.0.0, < 5.0.0"
          }
        }
      }

      provider "aiven" {
        api_token = var.aiven_api_token
      }
      ```

   - ``mysql.tf``, where you include the ``aiven_mysql`` resource

      ```hcl
      resource "aiven_mysql" "mysql" {
        project      = var.project_name
        service_name = var.service_name
        cloud_name   = var.cloud_name
        plan         = var.service_plan
      }

      output "mysql_service_host" {
        value = aiven_mysql.mysql.service_host
      }

      output "mysql_service_port" {
        value = aiven_mysql.mysql.service_port
      }

      output "mysql_service_username" {
        value = aiven_mysql.mysql.service_username
      }

      output "mysql_service_password" {
        value     = aiven_mysql.mysql.service_password
        sensitive = true
      }
      ```

>    - ``variables.tf``, where you declare the API token and project name variables

Reviewer: Since this is not only about those 2 variables, I think something more general would be better, e.g. ", where you declare your project variables."

      ```hcl
      variable "aiven_api_token" {
        description = "Aiven API token"
        type        = string
      }

      variable "project_name" {
        description = "Project name"
        type        = string
      }

      variable "cloud_name" {
        description = "Cloud name"
        type        = string
      }

      variable "service_name" {
        description = "Service name"
        type        = string
      }

      variable "service_plan" {
        description = "Service plan"
        type        = string
      }
      ```

>    - ``terraform.tfvars``, where you add the Aiven access token and project name

Reviewer: We do not assign a value for the API token here, so same here as well, I would just write, e.g. ", where you assign actual values to your previously declared variables."

      ```hcl
      project_name = "testproject-o3jb"
      cloud_name   = "google-europe-west3"
      service_name = "mysql"
      service_plan = "startup-4"
      ```

> 1. Store Terraform outputs in environment variables so that they can be used for

Reviewer: This needs to come after "Run ``terraform init`` > ``terraform plan`` > ``terraform apply --auto-approve``."

+    backup_hour      = 01
+    backup_minute    = 30
+    ip_filter_string = ["10.20.0.0/16"]
+    service_log      = true
+
+    mysql {
+      slow_query_log  = true
+      long_query_time = 5
+    }
+  }
}

output "mysql_service_host" {
  value = aiven_mysql.mysql.service_host
}

output "mysql_service_port" {
  value = aiven_mysql.mysql.service_port
}

output "mysql_service_username" {
  value = aiven_mysql.mysql.service_username
}

output "mysql_service_password" {
  value     = aiven_mysql.mysql.service_password
  sensitive = true
}
```

</TabItem>
</Tabs>

See the available configuration options in
[Advanced parameters for Aiven for MySQL](/docs/products/mysql/reference/advanced-params).

## Connect to the service{#connect-to-service}

<Tabs groupId="group1">
<TabItem value="1" label="Console" default>
1. Log in to the [Aiven Console](https://console.aiven.io/), and go to your
   organization > project > Aiven for MySQL service.
1. On the <ConsoleLabel name="overview"/> page of your service, click
   **Quick connect**.
1. In the **Connect** window, select a tool or language to connect to your service, follow
   the connection instructions, and click **Done**.

>    ```sql

Reviewer 2: ```suggestion
   ```bash
```

   mysql --user avnadmin --password=ADMIN_PASSWORD --host mysql-sakila-dev-sandbox.f.aivencloud.com --port 12691 defaultdb
   ```

</TabItem>
<TabItem value="2" label="Terraform">

Access your new service with the MySQL client using the environment variables assigned to
Terraform outputs:

> ```sql

Reviewer 2: ```suggestion
```bash
```

[mysql](https://dev.mysql.com/doc/refman/8.0/en/mysql.html).
</TabItem>
</Tabs>

:::tip
Check more tools for connecting to Aiven for MySQL in
[Connect to Aiven for MySQL](/docs/products/mysql/howto/list-code-samples).
:::

## Load a test dataset

`Sakila` is a sample dataset that represents a DVD rental store. It provides a standard
schema highlighting MySQL features.

1.  Download the `sakila` database archive (`tar` or `zip` format) from the
    [MySQL example databases](https://dev.mysql.com/doc/sakila/en/sakila-installation.html)
    page, and extract it to your desired location (for example `/tmp/`).

1.  From the folder where you unpacked the archive,
    [connect to your MySQL service](/docs/products/mysql/howto/connect-from-cli),
    create a `sakila` database, and connect to it:

    ```sql
    CREATE DATABASE sakila;
    USE sakila;
    ```

1.  Populate the database:

    ```sql
    source sakila-schema.sql;
    source sakila-data.sql;
    ```

1.  Verify what objects have been created:

    ```sql
    SHOW FULL TABLES;
    ```
    <!-- vale off -->
    <details>
    <summary>Expected output</summary>
    <div>

>         ```sql

Reviewer 2: CLI output is typically text:

```suggestion
        ```text
```

        | customer_list              | VIEW       |
        | film                       | BASE TABLE |
        | film_actor                 | BASE TABLE |
        | film_category              | BASE TABLE |
        | film_list                  | VIEW       |
        | film_text                  | BASE TABLE |
        | inventory                  | BASE TABLE |
        | language                   | BASE TABLE |
        | nicer_but_slower_film_list | VIEW       |
        | payment                    | BASE TABLE |
        | rental                     | BASE TABLE |
        | sales_by_film_category     | VIEW       |
        | sales_by_store             | VIEW       |
        | staff                      | BASE TABLE |
        | staff_list                 | VIEW       |
        | store                      | BASE TABLE |
        +----------------------------+------------+
        23 rows in set
        ```
    </div>
    </details>
    <!-- vale on -->

## Query data

### Read data

Retrieve all the data from a table, for example, from `language`:

```sql
SELECT * FROM language;
```

<!-- vale off -->
<details>
<summary>Expected output</summary>
<div>

>     ```sql

Reviewer 2: ```suggestion
    ```text
```

    +-------------+----------+---------------------+
    | language_id | name     | last_update         |
    +-------------+----------+---------------------+
    |           1 | English  | 2006-02-15 05:02:19 |
    |           2 | Italian  | 2006-02-15 05:02:19 |
    |           3 | Japanese | 2006-02-15 05:02:19 |
    |           4 | Mandarin | 2006-02-15 05:02:19 |
    |           5 | French   | 2006-02-15 05:02:19 |
    |           6 | German   | 2006-02-15 05:02:19 |
    +-------------+----------+---------------------+
    6 rows in set
    ```
</div>
</details>
<!-- vale on -->

### Write data

Add a row to a table, for example, to `category`:

```sql
INSERT INTO category(category_id,name) VALUES(17,'Thriller');
```

<!-- vale off -->
<details>
<summary>Expected output</summary>
<div>

>     ```sql

Reviewer 2: ```suggestion
    ```text
```

    Query OK, 1 row affected
    ```
</div>
</details>
<!-- vale on -->

Check that your new row is there:

```sql
SELECT * FROM category WHERE name = 'Thriller';
```

<!-- vale off -->
<details>
<summary>Expected output</summary>
<div>

>     ```sql

Reviewer 2: ```suggestion
    ```text
```
